idaes/surrogate/pysmo_surrogate.py

```diff
@@ -0,0 +1,772 @@
+        self.model_type = f"{self.config.basis_function} {self.base_model_type}"
+
+    def _create_model(self, pysmo_input, output_label):
+        model = rbf.RadialBasisFunctions(
+            pysmo_input,
+            basis_function=self.config.basis_function,
+            solution_method=self.config.solution_method,
+            regularization=self.config.regularization,
+            overwrite=True,
+        )
+        variable_headers = model.get_feature_vector()
+        return model
+
+    def _get_metrics(self, model) -> Dict:
+        return {"R2": model.R2, "RMSE": model.rmse}
+
+
+class PysmoKrigingTrainer(PysmoTrainer):
+    """Train a Kriging surrogate model."""
+
+    model_type = "kriging"
+
+    CONFIG = PysmoTrainer.CONFIG()
+
+    CONFIG.declare(
+        "numerical_gradients",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Choice of whether numerical gradients are used in Kriging model training."
+            "Determines choice of optimization algorithm: Basinhopping (False) or BFGS (True)."
+            "Using the numerical gradient option leads to quicker (but in complex cases possible sub-optimal)"
+            " convergence",
+        ),
+    )
+
+    CONFIG.declare(
+        "regularization",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Option for regularization - results in a regression rather than interpolation. "
+            "Produces more generalizable models. Useful for noisy data.",
+        ),
+    )
+
+    def __init__(self, **settings):
+        super().__init__(**settings)
+
+    def _create_model(self, pysmo_input, output_label):
+        model = krg.KrigingModel(
+            pysmo_input,
+            numerical_gradients=self.config.numerical_gradients,
+            regularization=self.config.regularization,
+            overwrite=True,
+        )
+        variable_headers = model.get_feature_vector()
+        return model
+
+    def _get_metrics(self, model):
+        return {"RMSE": model.training_rmse, "R2": model.training_R2}
+
+
+class PysmoSurrogate(SurrogateBase):
+    """PySMO surrogate model API."""
+
+    def __init__(
+        self,
+        trained_surrogates: TrainedSurrogate,
+        input_labels,
+        output_labels,
+        input_bounds=None,
+    ):
+        """Create PySMO surrogate model.
+        Args:
+            trained_surrogates: Results of training surrogates.
+            input_labels:
+            output_labels:
+            input_bounds:
+        """
+        super().__init__(input_labels, output_labels, input_bounds)
+        self._trained = trained_surrogates
+        self._trained.input_labels, self._trained.input_bounds = (
+            input_labels,
+            input_bounds,
+        )
+
+    def evaluate_surrogate(self, inputs: pd.DataFrame) -> pd.DataFrame:
+        """
+            Evaluate the surrogate model at a set of user-provided values.
```

Comment:
```suggestion
        """Evaluate the surrogate model at a set of user-provided values.
```

Reply:
The only thing I notice about the functions Sphinx is complaining about is the fact that they start from an indented new line. I'd try to remove the extra space (i.e. so that e.g. the docstring for `load()` is similar to the one for `save()`) and see if that fixes the issue.